src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpPossiblyDeclareAsNullableCodeFixProvider.cs

```diff
@@ -0,0 +1,140 @@
+    {
+        [ImportingConstructor]
+        public CSharpPossiblyDeclareAsNullableCodeFixProvider()
+        {
+        }
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(IDEDiagnosticIds.PossiblyDeclareAsNullableDiagnosticId);
+
+        // No support for FixAll at the moment. For example, in public API code this fix is likely incorrect.
+        public override FixAllProvider GetFixAllProvider() => null;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.First();
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+
+            var fixableType = await TryGetFixableTypeAsync(document, diagnostic, cancellationToken).ConfigureAwait(false);
+            if (fixableType is object)
+            {
+                context.RegisterCodeFix(new MyCodeAction(
+                    c => FixAsync(context.Document, diagnostic, c)),
+                    context.Diagnostics);
+            }
+        }
+
+        private async Task<TypeSyntax> TryGetFixableTypeAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+            var model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var symbolToFix = CSharpPossiblyDeclareAsNullableDiagnosticAnalyzer.IsFixable(node, model);
+
+            var declarationLocation = symbolToFix.Locations[0];
+            var typeNode = declarationLocation.FindNode(getInnermostNodeForTie: true, cancellationToken);
+            var typeToFix = TryGetTypeToFix(typeNode);
+
+            if (typeToFix == null || typeToFix is NullableTypeSyntax)
+            {
+                return null;
+            }
+
+            return typeToFix;
+        }
+
+        private async Task<Document> FixAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var node = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+
+            var model = await document.GetSemanticModelAsync().ConfigureAwait(false);
+
+            var symbolToFix = CSharpPossiblyDeclareAsNullableDiagnosticAnalyzer.IsFixable(node, model);
+            if (symbolToFix == null)
+            {
+                return document;
+            }
+
+            var declarationLocation = symbolToFix.Locations[0];
+            var typeNode = declarationLocation.FindNode(getInnermostNodeForTie: true, cancellationToken);
+
+            var typeToFix = TryGetTypeToFix(typeNode);
+            if (typeToFix == null || typeToFix is NullableTypeSyntax)
```

Comment:
all callers seem to need to check for NullableTypeSyntax.  Consider pushing that check inside so that the caller doesn't have to care.

Reply:
also, thi logic looks duplicated with TryGetFixableTypeAsync, can we not reuse that?